pkg/storage/stores/shipper/compactor/retention/index.go

```diff
@@ -109,14 +110,24 @@ func parseChunkID(chunkID []byte) (userID []byte, hexFrom, hexThrough []byte, va
 		hex = chunkID[j+1:]
 		break
 	}
+
 	if len(userID) == 0 {
 		return nil, nil, nil, false
 	}
-	_, i = readOneHexPart(hex)
-	if i == 0 {
-		return nil, nil, nil, false
+
+	// v12+ chunk id format `<user>/<fprint>/<start>:<end>:<checksum>`
```

Comment:
I think we should standardize the internal functions used to parse chunks in `pkg/storage/chunk` so we don't have to maintain two versions, but that can be left for a later pr.

Reply:
Yes ! +1 Let's add an issue.